### DISCUSSION TITLE
model_validate problem on Optional attribute using Field()

### DISCUSSION THREAD

**Discussion by lowang** [fastapi/sqlmodel#825] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import decimal
from typing import Annotated, Optional

from sqlmodel import Field, SQLModel

raw_data = {
    "c": decimal.Decimal("2.1"),
    "d": None,
}

# suggested by https://github.com/pydantic/pydantic/issues/7546
# produces exception TypeError: issubclass() arg 1 must be a class
# in sqlmodel/main.py line 567, in get_sqlalchemy_type
class TestTableProblem(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    c: decimal.Decimal
    d: Annotated[decimal.Decimal, Field(max_digits=20, decimal_places=10, nullable=True)] | None

TestTableProblem.model_validate(raw_data)
```

### Description

I'm using following features:
- `SQLModel.metadata.create_all(engine)`
- `CreateTable(model.__table__).compile(db_engine)`

when specifying attr like `decimal.Decimal` CREATE statement ignores decimal places (using MSSQL)

### Operating System

Linux, macOS

### Operating System Details

running it on docker with 
```
pydantic            2.6.2
pydantic_core       2.16.3
pydantic-settings   2.1.0
```

### SQLModel Version

0.0.16

### Python Version

3.10.12

### Additional Context

https://github.com/pydantic/pydantic/issues/7546
https://github.com/tiangolo/sqlmodel/discussions/791

**Comment by YuriiMotov** [comment, marked as answer]
```py
    d: Annotated[decimal.Decimal | None, Field(max_digits=20, decimal_places=10, nullable=True)]
```

The above works (move `| None` inside `Annotated`)


```
CREATE TABLE testtableproblem (
        id INTEGER NOT NULL AUTO_INCREMENT, 
        c NUMERIC NOT NULL, 
        d NUMERIC(20, 10), 
        PRIMARY KEY (id)
)
```
